internal/backend/remote-state/azure/auth/cli_auth.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+type AzureCLIAuthConfig struct {
+	CLIAuthEnabled bool
+}
+
+type azureCLICredentialAuth struct{}
+
+func (cred *azureCLICredentialAuth) Name() string {
+	return "Azure CLI Auth"
+}
+
+func (cred *azureCLICredentialAuth) Construct(_ context.Context, config *Config) (azcore.TokenCredential, error) {
+	// The SubscriptionID and TenantID can be empty, and the logic of this will still be okay
+	return azidentity.NewAzureCLICredential(&azidentity.AzureCLICredentialOptions{
+		Subscription: config.StorageAddresses.SubscriptionID,
+		TenantID:     config.StorageAddresses.TenantID,
+	})
+}
+func (cred *azureCLICredentialAuth) Validate(config *Config) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
```

Comment:
Do we want to try to detect if the az command is installed and disable CLI auth if not?  I'm not sure if that matches the existing functionality or not.

Reply:
Hmm, there's no cli setting for this.  I don't feel super strongly about it.  I'd suggest https://pkg.go.dev/os/exec#LookPath if we decided to do it.